file-sharing-back/file-sharing-back-core/src/main/java/com/file/sharing/core/service/impl/UserServiceImpl.java

```
@@ -44,4 +45,10 @@ public UserInfo getUserInfoByUserId(Integer userId) {
 		}
 		return userInfo.get();
 	}
+
+	@Override
+	public Subscription getUserSubscription(int userId) {
+		Optional<UserInfo> userInfo = userBusiness.getUserInfoByUserId(userId);
+		return userInfo.map(ui -> ui.getAccStatsInfo().getSubscription()).orElseThrow(() -> new UserNotFoundException(userId));
```

This is a bit strange... If the Subscription in the AccStatsInfo of the UserInfo is null, I throw an exception UserNotFoundException. We should throw this exception only if the user info is not found. If AccStatsInfo is not nullable, and also the subcription is not nullable, we dont' need to do any other check.
Code would look something like this:

`
Optional<UserInfo> userInfo = userBusiness.getUserInfoByUserId(userId);

if( ! userInfo.isPresnet()){
    throw UserNotFoundException();
}

return userInfo.map(ui -> ui.getAccStatsInfo().getSubscription()).orElseThrow(() -> 
SubscriptionNotFoundException());
`